src/create_mtc_pipeline_planner.cpp

```diff
@@ -31,38 +35,61 @@ BT::NodeStatus CreateMTCPipelinePlanner::tick()
-
-  // Build solver
+  if(!getInput(kPortPlannerID, planner_id))  //ex : "request_adapters"
+    return NodeStatus::FAILURE;
+  if(!getInput(kPortMaxVelocityScalingFactor, max_velocity_scaling_factor))
+    return NodeStatus::FAILURE;
+  if(!getInput(kPortMaxAccelerationScalingFactor, max_acceleration_scaling_factor))
+    return NodeStatus::FAILURE;
+  //Optionnal inputs
+  getInput(kPortGoalJointTolerance, goal_joint_tolerance);
+  getInput(kPortGoalPositionTolerance, goal_position_tolerance);
+  getInput(kPortGoalOrientationTolerance, goal_orientation_tolerance);
+  getInput(kPortDisplayMotionPlans, display_motion_plans);
+  getInput(kPortPublishPlanningRequests, publish_planning_requests);
+  getInput(kPortNumPlanningAttempts, num_planning_attemps);
+  //build solver
   auto solver = std::make_shared<MTC::solvers::PipelinePlanner>(pipeline_id);
   solver->setPlannerId(planner_id);
   solver->setMaxVelocityScalingFactor(max_velocity_scaling_factor);
   solver->setMaxAccelerationScalingFactor(max_acceleration_scaling_factor);
   solver->setProperty(kPortGoalJointTolerance, goal_joint_tolerance);
-
+  solver->setProperty(kPortGoalOrientationTolerance, goal_orientation_tolerance);
+  solver->setProperty(kPortGoalPositionTolerance, goal_position_tolerance);
+  solver->setProperty(kPortDisplayMotionPlans, display_motion_plans);
+  solver->setProperty(kPortPublishPlanningRequests, publish_planning_requests);
+  solver->setProperty(kPortNumPlanningAttempts, num_planning_attemps);
   // Upcast to base class
   MTC::solvers::PlannerInterfacePtr base_solver = solver;
 
   setOutput(kPortSolver, base_solver);
-
   return NodeStatus::SUCCESS;
 }
 
 BT::PortsList CreateMTCPipelinePlanner::providedPorts()
 {
   return {
-    BT::InputPort<std::string>(kPortPipelineID),
-    BT::InputPort<std::string>(kPortPlannerID),
-    BT::InputPort<double>(kPortGoalJointTolerance, 1e-4, "tolerance for reaching joint goals"),
-    BT::InputPort<double>(kPortMaxVelocityScalingFactor, 0.1, "scale down max velocity by this factor"),
-    BT::InputPort<double>(kPortMaxAccelerationScalingFactor, 0.1, "cale down max acceleration by this factor"),
-    BT::OutputPort<MTC::solvers::PlannerInterfacePtr>(kPortSolver),
+    //Output
+    BT::OutputPort<MTC::solvers::PlannerInterfacePtr>(kPortSolver, "{solver}", "Planner interface using pipeline motion solver"),
+    //Inputs
+    BT::InputPort<double>(kPortGoalJointTolerance, "1e-4", "tolerance for reaching joint goals"),
+    BT::InputPort<double>(kPortGoalPositionTolerance, "1e-4", "tolerance for reaching position goals"),
+    BT::InputPort<double>(kPortGoalOrientationTolerance, "1e-4", "tolerance for reaching orientation goals"),
+    BT::InputPort<bool>(kPortDisplayMotionPlans, false, "..."),
+    BT::InputPort<bool>(kPortPublishPlanningRequests, false, "..."),
+    BT::InputPort<std::string>(kPortPlannerID, "planner's id"),
+    BT::InputPort<std::string>(kPortPipelineID, "pipeline's id"),
```

Comment:
No need to add a description as it is pretty self explanatory. At least for seasoned MoveIt users ;)
```suggestion
    BT::InputPort<std::string>(kPortPlannerID),
    BT::InputPort<std::string>(kPortPipelineID),
```